### PR TITLE
docs: update CONTRIBUTING.md regarding deprecation of `develop`

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR updates CONTRIBUTING.md regarding the deprecation of `develop` branch.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -210,7 +208,7 @@ If you're on a feature branch (either for a fix or a new minor/major version), y
 npm run release
 ```
 
-You can also release a beta version from `develop`, using the beta flag in the command line:
+You can also release a beta version from `next`, using the beta flag in the command line:
```
Comment (eunjae-lee) [inline CONTRIBUTING.md]: `next` instead of `develop` makes sense here?

Comment (francoischalifour) [reply CONTRIBUTING.md]: `next` makes sense for the next major version, we can add a mention in "Branch organization".

However, I think this sentence doesn't make sense anymore. We should be able to release a beta version from any branch with Ship.js.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -182,7 +180,7 @@ JavaScript and TypeScript files are validated using a combination of [Prettier](
 
 ### Main version
 
-For the main version, go on develop (`git checkout develop`) and use:
+For the main version, go on master (`git checkout master`) and use:
```
Comment (francoischalifour) [inline CONTRIBUTING.md]:
```suggestion
For the main version, go on `master` (`git checkout master`) and use:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -61,7 +61,6 @@ After you create a pull request, a bot will comment with a link to a development
 You will find a playground for the widgets: https://**netlify-deploy-url**/stories.
 
 - Example: https://deploy-preview-3376--instantsearchjs.netlify.com/stories/
```
Comment (francoischalifour) [inline CONTRIBUTING.md]: If there's now only one item in the list, we can inline it in the line above.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Haroenv) [reply CONTRIBUTING.md]:
```suggestion
You can also release a beta version from another branch, using the beta flag in the command line:
```

?

File: CONTRIBUTING.md (hunk 3 above)
Comment (eunjae-lee) [reply CONTRIBUTING.md]: https://github.com/algolia/instantsearch.js/pull/4280/commits/251433a6a8e6e6a71a1585a3f1c9e8fa4625e449

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -210,7 +206,7 @@ If you're on a feature branch (either for a fix or a new minor/major version), y
 npm run release
 ```
 
-You can also release a beta version from `develop`, using the beta flag in the command line:
+You can also release a beta version from another branch, using the beta flag in the command line:
```
Comment (francoischalifour) [inline CONTRIBUTING.md]:
```suggestion
You can release beta versions from any branches, using the beta flag in the command line:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -182,7 +178,7 @@ JavaScript and TypeScript files are validated using a combination of [Prettier](
 
 ### Main version
 
-For the main version, go on develop (`git checkout develop`) and use:
+For the main version, go on `master` (`git checkout master`) and use:
```
Comment (francoischalifour) [inline CONTRIBUTING.md]:
```suggestion
To release a stable version, go on `master` (`git checkout master`) and use:
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (Haroenv) [reply CONTRIBUTING.md]:
```suggestion
You can release beta versions from any branch, using the beta flag in the command line:
```

**Comment by eunjae-lee** [comment]
@yannickcr the e2e test keeps failing 😢 I re-ran it, but it's still the same.
Anyway, this PR is just for a `CONTRIBUTIN.md` file, so I will merge this regardlessly.

**Comment by yannickcr** [comment]
@eunjae-lee no problem, I think that we can say that your change is safe here :wink: 

Sorry for the flaky tests :cry:

**Pull request merged**